pkg/proc/interface.go

```diff
@@ -115,8 +115,19 @@ type BreakpointManipulation interface {
 // implementations of the Process interface.
 type CommonProcess struct {
 	allGCache     []*G
-	fncallState   functionCallState
 	fncallEnabled bool
+
+	// if continueCompleted is not nil it means we are in the process of
+	// executing an injected function call, see comments throughout
+	// pkg/proc/fncall.go for a description of how this works.
+	continueCompleted chan<- struct{}
+	continueRequest   <-chan continueRequest
+
+	// callInProgress is true when a function call is being injected in the
+	// target process.
+	// This is only used to prevent nested function calls, it should be removed
+	// when we add support for them.
+	callInProgress bool
```

Comment:
continueCompleted is initialized by EvalExpressionWithCalls and it means that we are allowed to call functions. callInProgress means that we started evaluating a function already and we can't start another one. 

If the user trys to evaluate `f(g())` through EvalExpression (so a `print f(g())`) the code in fncall.go will find `continueCompleted == nil` and tell the user that calls are not allowed.

By contrast if the command is `call f(g())` the call to `f` will be allowed (because `continueCompleted != nil`) but the nested call to `g` won't because `callInProgress == true`.

This variable gets removed by [the next function call PR](https://github.com/go-delve/delve/pull/1504).